### PR TITLE
fix(notifications): disable sounds on DND

### LINKED ISSUE #415: notification sounds do not respect `Do Not Disturb` user status
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Set user profile status to `Do Not Disturb` (via web.. status changes not yet supported.. #26 ) 
2. Receive new message
3. Notification sound plays

### Expected behaviour
Expect the notification sounds to be automatically muted while the user status is set to DND. 

### Actual behaviour
notification sounds play despite the user status is set to DND. 

## Desktop client

```
----------------------------System report----------------------------
Nextcloud Talk Desktop version 0.14.0
- Built with Nextcloud Talk version 18.0.0-beta.1

Connected to:
- Server address: https://cloud.nextcloud.com
- Nextcloud Server version 28.0.0 beta 2
- Nextcloud Talk version 18.0.0-beta.2

OS: Linux 6.5.9-1-default (#1 SMP PREEMPT_DYNAMIC Wed Oct 25 10:31:37 UTC 2023 (29edc7c))
----------------------------System report----------------------------
```

**Operating system:** Windows / MacOS / Ubuntu
openSUSE Tumbleweed - KDE Plasma

**Operating system version:** Windows 11 / (In case of Linux, provide the distribution.)

**Microphone available:** yes/no

**Camera available:** yes/no

## Server

**Nextcloud version:** (see status page: `/status.php`)

**Talk app version:** (see apps admin page: `/index.php/settings/apps`)

**Custom Signaling server configured:** yes/no and version (see additional admin settings: `/index.php/index.php/settings/admin/talk#signaling_server`)

**Custom TURN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#turn_server`)

**Custom STUN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#stun_server`)

### Client log

<details>

```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

* Fix #415

### 🚧 Tasks

- [x] Add utils to set `InitialState`
- [x] On user status update change `playSound` configuration of `notifications` app
  - Defined by initial state
- [x] Also disable native notification sound as we always have out custom sound

### CODE REVIEW

File: src/talk/renderer/notifications/notifications.store.js [hunk 1]
```diff
@@ -11,13 +11,31 @@
...
+ * Enables or disables the sound for notifications and calls in the initial state
+ * @param {boolean} playSound - Whether to enable play a sound
+ */
+const setPlaySoundState = (playSound) => {
+	setInitialState('notifications', 'sound_notification', playSound)
+	setInitialState('notifications', 'sound_talk', playSound)
+}
+
+// Listen for user status changes to disable sounds on DND and enable it back otherwise
+subscribe('user_status:status.updated', (userStatus) => {
```
Comment (nickvergessen) [inline src/talk/renderer/notifications/notifications.store.js:33]: Does the status event trigger initially? Or what happens when you start the client while you're already on DND?

Comment (ShGKme) [reply src/talk/renderer/notifications/notifications.store.js:33]: Well, yes and no.

Initially, when it is restored from the cache — no.

In 1 second when it is requested from the server — yes.

Makes sense to also emit initially?

Comment (nickvergessen) [reply src/talk/renderer/notifications/notifications.store.js:33]: well 1s is good enough. Just don't want to wait until it actually changed.

Comment (ShGKme) [reply src/talk/renderer/notifications/notifications.store.js]: Done anyway, it seems better to use the status initially. Like if GET user status request is slower than loading new notifications.

**Review by DorraJaouad — APPROVED** [review]
LGTM

**Comment by ShGKme** [comment]
Rebased onto `main` and squashed

**Pull request merged**